app/scripts/views/customizer/Widgets/Geometry.js

```diff
@@ -1,225 +1,7 @@
-
-      for ( var i = 0; i < rawGeom.vertices.length; i++ ) {
-        var v = rawGeom.vertices[i];
-        threeGeom.vertices.push( new THREE.Vector3( v[0], v[1], v[2] ) );
-      }
-
-      threeGeom._floodType = 1;
-
-      return threeGeom;
-    },
-
-    addLineStrip: function( rawGeom, threeGeom ){
-
-      for ( var i = 0; i < rawGeom.linestrip.length; i++ ) {
-        var v = rawGeom.linestrip[i];
-        threeGeom.vertices.push( new THREE.Vector3( v[0], v[1], v[2] ) );
-      }
-
-      threeGeom._floodType = 2;
-
-      return threeGeom;
-      
-    },
-
-    onEvalComplete: function(a, b, newValue){
-
-      if (!newValue && this.evaluated) return;
-
-      this.evaluated = true;
-
-      var lastValue = this.model.get('prettyLastValue');
-      var temp;
-
-      if ( !lastValue ) return;
-
-      if ( lastValue.vertices || lastValue.linestrip ){ 
-        temp = [];
-        temp.push(lastValue);
-      } else {
-        temp = lastValue; // extract the list
-      } 
-
-      var threeTemp = new THREE.Object3D();
-      this.drawChunked( threeTemp, temp, function() { 
-
-        if ( this.threeGeom ){
-          scene.remove( this.threeGeom );
-        }
-
-        this.threeGeom = threeTemp;
-        scene.add( this.threeGeom );
-        this.changeVisibility();
-
-      }, this );
-
-    }, 
-
-    // creating this data may be quite slow, we'll need to be careful
-    drawChunked: function(geom, list, callback, that){
-
-      var i = 0;
-      var tick = function() {
-
-        var start = new Date().getTime();
-        for (; i < list.length && (new Date().getTime()) - start < 50; i++) {
-        
-          var g3  = that.toThreeGeom( list[i] );
-
-          if (that.model.get('selected')){
-            var color = 0x66d6ff;
-          } else {
-            var color = 0x999999;
-          }
-
-          switch (g3._floodType) {
-            case 0:
-              geom.add( new THREE.Mesh(g3, new THREE.MeshPhongMaterial({color: color, side : THREE.DoubleSide})) );
-              break;
-            case 1:
-              geom.add( new THREE.ParticleSystem(g3, new THREE.ParticleBasicMaterial({color: color, size: 5, sizeAttenuation: false}) ));
-              break;
-            case 2:
-              geom.add( new THREE.Line(g3, new THREE.LineBasicMaterial({ color: 0x000000 })));
-              break;
-          }
-
-        }
-
-        if (i < list.length) {
-          setTimeout(tick, 25);
-        } else {
-          callback.call(that);
-        }
-
-      };
-
-      setTimeout(tick, 0);
-
-    },
-
-    changeVisibility: function(){
-
-      if ( !this.threeGeom ){
-        return;
-      }
-        
-      if (!this.model.get('visible') || !this.model.workspace.get('current') )
-      {
-        this.threeGeom.traverse(function(e) { e.visible = false; });
-      } else if ( this.model.get('visible') )
-      {
-        this.threeGeom.traverse(function(e) { e.visible = true; });
```

Comment:
@pbidenko, It turns out that flipping this bit is far faster than removing Geometry from the Scene.  We should stick to this model.